### ISSUE TITLE
Training checklist

### ISSUE THREAD

**Issue by justheuristic** [NCAI-Research/CALM#1]
### Start some peers
- [x] @razaidy starts initial CPU peer and shares her Peer ID  (/ip4/something/tcp/something)
- [x] @justheuristic starts another CPU peer for fault-tolerance and shares his Peer ID
- [x] Start a few GPU workers

### Technical features
- [x] Scale sequence length dynamically during training [@justheuristic]
   - add 32 every 1000 global steps up to 512
- [ ] training with TPU

### Volunteer starter kit
- [x] [draft notebook](https://colab.research.google.com/drive/1Y74o9UvJypE185nNiDcrV6PHJ7ud3XyP#scrollTo=w1d4ZenoXmPu)
- [x] create an invite link for volunteers
- [x] text explanation for running in colab
- [x] upload notebook to `CALM/notebooks`?
- [x] discord chatroom
- [x] organization page
- [x] instructions for training locally, with kaggle, with sagemaker
- [x] email to volunteers (early ~wednesday )

### After we start
- [x] contributors dashboard [@SaulLu ]
- [ ] running evaluation every few hours
- [ ] set up monitoring & support shifts

### Sanity checks
- [x] make sure training data looks right (@razaidy @JAWHARAH123 @pr-Mais )
- [x] make sure loss is in a reasonable range 
- [x] verify that volunteer starter kits work
- [ ] make sure training with TPU does not leak memory during host<->device transfer (@justheuristic )
- [x] look at the data again just in case

### Milestones
- [x] pass loss 9 by ~1000 steps (without stagnating)
![image](https://user-images.githubusercontent.com/3491902/147371558-41c9c9e4-3e76-4091-b7c1-b320165a7519.png)
- [x] do not blow up at peak learning rate (steps 2500-3500)
- [x] downstream should be better than random after step 4000
- [ ] reach full sequence length (10000)

**Comment by pr-Mais** [comment]
I see sagemaker, could GCE w or w/o colab be used as well?

**Comment by justheuristic** [comment]
Anything can as long as it has a free tier :)
Sagemaker had a __free__ studiolab: https://studiolab.sagemaker.aws/

**Comment by SaulLu** [comment]
Some news about the dashboard :smiley: 

Here is what I did for the moment: 
1. Create a streamlit space named [Dashboard](https://huggingface.co/spaces/CALM/Dashboard) in the CALM organization
2. Copy the [dashboard repo of training-transformers-together](https://github.com/training-transformers-together/dashboard) into [the new github repo for the CALM dashboard](https://github.com/NCAI-Research/dashboard)
3. Change the target HF repository in the github workflow ([here](https://github.com/NCAI-Research/dashboard/blob/main/.github/workflows/sync_to_hub.yaml#L20))
5. Create a [machine-user](https://huggingface.co/ncai-calm-machine-user) for the organization and add it to the organization with a WRITE role (we need its token) - how should I share with you its password?

To finish setting up this dashboard, we should : 
6. Add a secret named `HF_TOKEN` in the [dashboard repository on github ](https://github.com/NCAI-Research/dashboard) corresponding to a `write` access token of the [machine user](https://huggingface.co/ncai-calm-machine-user) (I don't have the necessary rights on github)
7. Add two secrets `WANDB_REPO_INDIVIDUAL_METRICS` and `WANDB_RUN_URL_MAIN_METRICS` in the [dashboard repository on the HUB ](https://huggingface.co/spaces/CALM/Dashboard) corresponding to the links to the WANDBs storing the data.

**Comment by justheuristic** [comment]
Awesome work!

For step #6
@razaidy , you mentioned there is someone on your side who can admin the organization, right?

@SaulLu can they use [this instruction](https://docs.github.com/en/actions/security-guides/encrypted-secrets#creating-encrypted-secrets-for-a-repository) for adding a secret?

For step #7, I took the liberty to add these two secrets, but I'm not entirely sure I got the format right. Since they are public knowledge, I'll copy them here as well:
- `WANDB_REPO_INDIVIDUAL_METRICS`=`https://wandb.ai/calm/CALM-hivemind-trainers`
- `WANDB_RUN_URL_MAIN_METRICS`=`https://wandb.ai/calm/CALM`

p.s. this seems way more elegant than what we did with Neuropark (i.e. hardcode access tokens), thanks!

**Comment by JAWHARAH123** [comment]
Thank you SaulLu for your effort and contribution.

Can you please provide me with the write access token of the machine user? 
so I can add the secret to the repository.

**Comment by SaulLu** [comment]
@JAWHARAH123 , with pleasure to share with you this secret information (I'll share the password to this machine user and the user access token)! I think that I can send you a private message on github, How do you want me to reach you? (email? discord?) :smile: 

The procedure to add the secret to the repository is indeed exactly the one you shared @justheuristic !

Thanks for the information regarding the step 7 @justheuristic  :hugs:

**Comment by JAWHARAH123** [comment]
Whatever you like ,you can find me in the CALM discord channel, or you can
send me an email ***@***.***)

في سبت، ١ يناير، ٢٠٢٢ في ٦:١١ م، كتب SaulLu ***@***.***>:

**Comment by SaulLu** [comment]
Thank you very much for your answer, unfortunately I can't see your email address in your last message and there are more users with the same nickname as you on discord.
![image](https://user-images.githubusercontent.com/55560583/147862094-e780e618-9b43-4c03-abec-e6b6c4813f5e.png)

If it ever helps, on discord I am user SaulLu `#0201`.

**Comment by SaulLu** [comment]
Thank you all! The dashboard is live at this address: [https://hf.co/spaces/CALM/Dashboard](https://huggingface.co/spaces/CALM/Dashboard)
